### ISSUE TITLE
TestCronSuite/TestBasic/TestBasicReplace is flakey

### ISSUE THREAD

**Issue by simster7** [argoproj/argo#4171]
```
2020-09-29T15:21:44.4255822Z Checking cron expectation
2020-09-29T15:21:44.4256758Z --- FAIL: TestCronSuite (308.85s)
2020-09-29T15:21:44.4259066Z     cron_test.go:294: 
2020-09-29T15:21:44.4259774Z         	Error Trace:	cron_test.go:294
2020-09-29T15:21:44.4260294Z         	            				then.go:71
2020-09-29T15:21:44.4260812Z         	            				cron_test.go:293
2020-09-29T15:21:44.4261290Z         	            				suite.go:77
2020-09-29T15:21:44.4261770Z         	Error:      	Not equal: 
2020-09-29T15:21:44.4262250Z         	            	expected: 1
2020-09-29T15:21:44.4263182Z         	            	actual  : 2
2020-09-29T15:21:44.4263718Z         	Test:       	TestCronSuite
2020-09-29T15:21:44.4264636Z     --- FAIL: TestCronSuite/TestBasic (0.00s)
2020-09-29T15:21:44.4265659Z         --- FAIL: TestCronSuite/TestBasic/TestBasicReplace (140.07s)
2020-09-29T15:21:44.4268691Z ##[error]            testing.go:864: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
2020-09-29T15:21:44.4273324Z FAIL
2020-09-29T15:21:44.4273922Z FAIL	github.com/argoproj/argo/test/e2e	308.894s
2020-09-29T15:21:44.4274415Z FAIL
2020-09-29T15:21:44.4783370Z ##[error]Makefile:396: recipe for target 'test-e2e-cron' failed
2020-09-29T15:21:44.4785396Z make: *** [test-e2e-cron] Error 1
2020-09-29T15:21:44.4792962Z ##[error]Process completed with exit code 2.
2020-09-29T15:21:44.4860586Z ##[group]Run actions/upload-artifact@v1
2020-09-29T15:21:44.4860990Z with:
2020-09-29T15:21:44.4861442Z   name: test-e2e-cron-278781591-argo.log
2020-09-29T15:21:44.4861965Z   path: /tmp/log/argo-e2e/argo.log
2020-09-29T15:21:44.4862291Z env:
2020-09-29T15:21:44.4862661Z   GOROOT: /opt/hostedtoolcache/go/1.13.12/x64
2020-09-29T15:21:44.4865639Z ##[endgroup]
2020-09-29T15:21:44.7491375Z Uploading artifact 'test-e2e-cron-278781591-argo.log' from '/tmp/log/argo-e2e/argo.log' for run #3098
```

**Comment by simster7** [comment]
```
[32m controller | [mtime="2020-09-29T15:19:00Z" level=warning msg="Workflow 'test-cron-wf-basic-replace-9w57k' from CronWorkflow 'test-cron-wf-basic-replace' completed"
[32m controller | [mtime="2020-09-29T15:19:00Z" level=error msg="cannot update cron workflow with mismatched resource versions" namespace=argo workflow=test-cron-wf-basic-replace
```

**Comment by alexec** [comment]
https://github.com/argoproj/argo/pull/4401/checks?check_run_id=1316404471

**Comment by alexec** [comment]
@simster7 can I please ask you bump the priority of this please? It seems to fail about 25% of builds.

This should obviously be lower priority to the fix for duplicate cron workflows.

**Comment by simster7** [comment]
Absolutely, I hope to have cron workflows done today, I will work on this next.

**Comment by alexec** [comment]
Maybe they are related. Hopefully!

**Comment by simster7** [comment]
> Maybe they are related. Hopefully!

I am now 90% sure they are. This should be fixed with #4440
